src/api/java/mekanism/api/gear/ModuleData.java

```diff
@@ -275,4 +283,20 @@ public ModuleDataBuilder<MODULE> disabledByDefault() {
             return this;
         }
     }
+
+    public enum ExclusiveFlag {
+        INTERACT_EMPTY,
+        INTERACT_ENTITY,
+        INTERACT_BLOCK,
+        OVERRIDE_JUMP;
+
+        // This will work for 32 flags, if we go past that we will have to switch it to a long. But that is a lot of flags!
+        public static int getFlags(ExclusiveFlag... flags) {
+            return Arrays.stream(flags).mapToInt(Enum::ordinal).reduce(0, (result, flag) -> result | (1 << flag));
```

Comment:
Are you talking about for performance concerns or readability? As this is only called for registration, I don't know that performance is really an issue. I've done flags explicitly and with constants and never liked how it looked, and it was always annoying to add/swap orders on the items. There are a couple things about this implementation that I really like, it's clear what the flags are and how to add another, getting compound masks is also easy without messing with bitwise yourself and we can also change the data type relatively easily.

But you're right adding a getMask function is a good idea, we just don't need to add it explicitly to the enum values.

Reply:
Readability in terms of seeing what corresponds to which. Also given the fact this is exposed as API (so should also have javadocs, the specific flags should also have javadocs briefly explaining them) you cannot state this is only used for initialization as it may be used at any time.

Reply:
True, I only meant that it's currently only called then. But it's not like it's expensive anyway.

With the getMask function added what do you think about keeping at as a basic enum?

Reply:
If you mean keeping it as a basic enum vs just making it a bunch of static ints.. I am unsure part of me feels like ints would be a bit less expensive to read than getting the ordinal and doing a bit shift each time you want to see if something conflicts with it, but I am open to either for now at least until there is some use case where it ends up as a hotspot.